src/commonMain/kotlin/com/algolia/search/configuration/Authentication.kt

```diff
@@ -0,0 +1,22 @@
+package com.algolia.search.configuration
+
+import com.algolia.search.model.APIKey
+import com.algolia.search.model.ApplicationID
+
+
+/**
+ * Configuration used by a client for authenticated request.
+ */
+public interface Authentication {
```

Comment:
`Authentication` sounds like something that authenticates rather than data object, WDYT of renaming to `Credentials`?

Reply:
Agreed